src/components/vm/consoles/serial.tsx

```diff
@@ -161,11 +171,21 @@ export const SerialMissing = ({
     );
 };
 
-export const SerialPending = () => {
+export const SerialPending = ({
+    vm,
+} : {
+    vm: VM,
+}) => {
+    const Dialogs = useDialogs();
+
     return (
         <EmptyState icon={PendingIcon} status="custom">
             <EmptyStateBody>
-                {_("Restart this virtual machine to access its serial console")}
+                <TextWithAction
+                    text={_("$0 this virtual machine to access its serial console")}
+                    action={_("Restart")}
+                    onClick={() => Dialogs.show(<VmRestartDialog vm={vm} />)}
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-2d5e2d69-20251027-143116-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)